### PR TITLE
feat: add grpc-web support #102

### PR DESCRIPTION
**Pull request by Avarei**
### Description of your changes

Implements #102

Adds optional fields `GRPCWeb` and `GRPCWebRootPath` to providerconfig and hands them over to the argocd client library.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Created a cluster object in an argocd behind a proxy without gRPC support.
Created another cluster object in an argocd within the same cluster over a k8s service (grpcWeb set to false).

I was not yet able to test GRPCWebRootPath

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/v1alpha1/providerconfig_types.go [hunk 1]
```diff
@@ -35,6 +35,14 @@ type ProviderConfigSpec struct {
 	// +optional
 	Insecure *bool `json:"insecure,omitempty"`
 
+	// Enables gRPC-web protocol. Useful if Argo CD server is behind proxy which does not support HTTP2.
+	// +optional
+	GRPCWeb         *bool `json:"grpcWeb,omitempty"`
+
+	// Enables gRPC-web protocol. Useful if Argo CD server is behind proxy which does not support HTTP2. Set web root.
+	// +optional
+	GRPCWebRootPath string `json:"grpcWebRootPath,omitempty"`
```
Comment (MisterMX) [inline apis/v1alpha1/providerconfig_types.go:44]: Optional fields should be a pointer

```suggestion
	GRPCWebRootPath *string `json:"grpcWebRootPath,omitempty"`
```

File: pkg/clients/argocd.go [hunk 2]
```diff
@@ -80,11 +80,18 @@ func UseProviderConfig(ctx context.Context, c client.Client, mg resource.Managed
 	if err != nil {
 		return nil, err
 	}
+	grpcWeb := false
+	if pc.Spec.GRPCWeb != nil {
+		grpcWeb = *pc.Spec.GRPCWeb
+	}
```
Comment (MisterMX) [inline pkg/clients/argocd.go:86]: Can be simplied with https://pkg.go.dev/k8s.io/utils@v0.0.0-20230726121419-3b25d923346b/ptr#Deref

```suggestion
	grpcWeb := ptr.Deref(pc.Spec.GRPCWeb, false)
```

**Review by MisterMX — CHANGES_REQUESTED** [review]

**Comment by Avarei** [comment]
I implemented your feedback and successfully ran `make reviewable` again.
I did not test these changes against an argocd yet. I will do that on monday.

**Comment by Avarei** [comment]
For me everything is working as expected in my cluster and agains my argocd

**Review by MisterMX — APPROVED** [review]
LGTM. Thank you very much @Avarei!

**Pull request merged**